lib/daru/dataframe.rb

```diff
@@ -1272,10 +1276,10 @@ def tail quantity=10
 
     # Returns a vector with sum of all vectors specified in the argument.
     # If vecs parameter is empty, sum all numeric vector.
-    def vector_sum vecs=nil
+    def vector_sum vecs=nil, skipnil=false
       vecs ||= numeric_vectors
       sum = Daru::Vector.new [0]*@size, index: @index, name: @name, dtype: @dtype
-
+      replace_nils(0) if skipnil
```

Comment:
Unfortunately, there is a problem with this approach. `replace_nils(0)` (as well as `replace_values`) changes the source DF. The result would be the following:
```ruby
df = Daru::DataFrame.new({
        a: [1,    2,          3,   nil,        Float::NAN, nil, 1,   7],
        b: [:a,  :b,          nil, Float::NAN, nil,        3,   5,   8],
        c: ['a',  Float::NAN, 3,   4,          3,          5,   nil, 7]
    })
# => #<Daru::DataFrame(8x3)>
#       a   b   c
#   0   1   a   a
#   1   2   b NaN
#   2   3 nil   3
#   3 nil NaN   4
#   4 NaN nil   3
#   5 nil   3   5
#   6   1   5 nil
#   7   7   8   7 

df.vector_sum(nil, true)
# => #<Daru::Vector(8)>
#   0   1
#   1   2
#   2   3
#   3   0
#   4 NaN
#   5   0
#   6   1
#   7   7 

df
# => #<Daru::DataFrame(8x3)>
#       a   b   c
#   0   1   a   a
#   1   2   b NaN
#   2   3   0   3
#   3   0 NaN   4
#   4 NaN   0   3
#   5   0   3   5
#   6   1   5   0
#   7   7   8   7 
```

Reply:
@zverok You're right, can fix by making it clone and sum or we can go with @parthm's approach. What do you think?